crates/swc_common/src/errors/diagnostic_builder.rs

```diff
@@ -27,7 +27,10 @@ use tracing::debug;
 #[derive(Clone)]
 pub struct DiagnosticBuilder<'a> {
     pub handler: &'a Handler,
+    #[cfg(not(feature = "plugin-mode"))]
     pub(crate) diagnostic: Box<Diagnostic>,
+    #[cfg(feature = "plugin-mode")]
+    pub diagnostic: Box<Diagnostic>,
```

Comment:
Exposing this for https://github.com/swc-project/swc/pull/3478/files#diff-4912e1065d4d9582db46d3a9a3b1ab8319c874737534bcaa725ff35f8b75b0d9R18 only to let plugin send diagnostic created via Handler into host.